Assets/Scripts/Capabilities/Jump.cs

```diff
@@ -72,26 +72,37 @@ private void FixedUpdate()
             coyoteTimeCounter -= Time.fixedDeltaTime;
         }
 
-        if(inputManager.GetInputRequested(InputManager.Input.Jump))
+        bool isJumpButtonHeld = inputController.RetrieveJumpButtonHeld();
+
+        if (inputManager.GetInputRequested(InputManager.Input.Jump))
         {
             JumpAction();
         }
-        if (body.velocity.y > 0)
+        if (body.velocity.y > 0 && isJumpButtonHeld)
         {
             body.gravityScale = movement.upwardMovementMultiplier;
         }
+        else if (body.velocity.y > 0 & !isJumpButtonHeld)
+        {
+            body.gravityScale = movement.upwardMovementShortJumpMultiplier;
+        }
         else if (body.velocity.y < 0)
         {
             body.gravityScale = movement.downwardMovementMultiplier;
         }
-        else if (body.velocity.y == 0)
+        else
         {
             body.gravityScale = movement.defaultGravityScale;
         }
 
         body.velocity = velocity;
 
-
+        // clamp fall speed to terminal velocity
+        if (body.velocity.y < 0)
+        {
+            float clampedYSpeed = Mathf.Clamp(body.velocity.y, -Constants.GRAVITY * movement.terminalVelocity, 0);
+            body.velocity = new Vector2(body.velocity.x, clampedYSpeed);
```

Comment:
Minor thing, can we set body.velocity outside of the if block and then just remove the earlier line so we're not setting it twice?

Reply:
We def can - is there a larger reason than code organization for this? e.g. if calling `body.velocity.set` does some extra internal stuff that would make this less performant?

Reply:
Just organization